pkg/controller/kubelet-config/helpers.go

```diff
@@ -97,8 +99,55 @@ func findKubeletConfig(mc *mcfgv1.MachineConfig) (*ign3types.File, error) {
 	return nil, fmt.Errorf("Could not find Kubelet Config")
 }
 
-func getManagedKubeletConfigKey(pool *mcfgv1.MachineConfigPool, client mcfgclientset.Interface) (string, error) {
-	return ctrlcommon.GetManagedKey(pool, client, "99", "kubelet", getManagedKubeletConfigKeyDeprecated(pool))
+// nolint: dupl
+func getManagedKubeletConfigKey(pool *mcfgv1.MachineConfigPool, client mcfgclientset.Interface, cfg *mcfgv1.KubeletConfig) (string, error) {
+	// Get all the kubelet config CRs
+	kcList, err := client.MachineconfigurationV1().KubeletConfigs().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return "", fmt.Errorf("error listing kubelet configs: %v", err)
+	}
+	// If there is no kubelet config in the list, return the default MC name with no suffix
+	if kcList == nil || kcList.Items == nil {
```

Comment:
`kcList.Items == nil` should be `len(kcList.Items) == 0`

Reply:
Fixed